### PR TITLE
SQL: Added Mob Skill for P. Warden / Lamps

### PR DESCRIPTION
**Pull request by Loki79**

### CODE REVIEW

**Comment by xdemolish** [comment]
perfect! thanks :D

**Comment by Loki79** [comment]
removed the custom mods in PW AI.

File: scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
--- Area: 
+-- Area:
 -- NPC:  Pandemonium_Warden
 -----------------------------------
-
 require("scripts/globals/titles");
+require("scripts/globals/status");
+require("scripts/globals/magic");
+require("scripts/globals/utils");
+require("scripts/globals/spoofchat");
```
Comment (xdemolish) [inline scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua]: hay thar custom content

File: scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua [hunk 2]
```diff
@@ -240,6 +185,12 @@ end;
...
 function onMobDeath(mob,killer)
-	-- TODO: Death speech.
-	killer:addTitle(PANDEMONIUM_QUELLER);
+    local CHANCE = 20;
+    if (math.random(0,99) < CHANCE and killer:getMainJob() == JOB_BLU and killer:hasSpell(742) == false) then
+        killer:addSpell(SPELL_ID);
+    end
+
+    -- mob:SpoofChatParty("This is the end..of nothing..Defeating me proves only that you too, harbor", MESSAGE_SAY)
+    -- mob:SpoofChatParty("seeds of evil within your trembling flesh..Someday..they shall come..into fruition..", MESSAGE_SAY)
```
Comment (xdemolish) [inline scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua]: we meet again

File: scripts/zones/AlTaieu/mobs/Jailer_of_Justice.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            mob:setLocalVar("pop_xzomit", os.time());
+        elseif (xzomit4 == ACTION_NONE or xzomit4 == ACTION_SPAWN) then
+            SpawnMob(mob:getID()+4, 300):updateEnmity(target);
+            mob:setLocalVar("pop_xzomit", os.time());
+        elseif (xzomit5 == ACTION_NONE or xzomit5 == ACTION_SPAWN) then
+            SpawnMob(mob:getID()+5, 300):updateEnmity(target);
+            mob:setLocalVar("pop_xzomit", os.time());
+        elseif (xzomit6 == ACTION_NONE or xzomit6 == ACTION_SPAWN) then
+            SpawnMob(mob:getID()+6, 300):updateEnmity(target);
+            mob:setLocalVar("pop_xzomit", os.time());
```
Comment (xdemolish) [inline scripts/zones/AlTaieu/mobs/Jailer_of_Justice.lua]: could just stick this in a for loop like
for i = mob:getID(), mob:getID() + 6 do
-- etc
end

File: scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua [hunk 4]
```diff
@@ -240,6 +185,12 @@ end;
 -----------------------------------
 
 function onMobDeath(mob,killer)
-	-- TODO: Death speech.
-	killer:addTitle(PANDEMONIUM_QUELLER);
+    local CHANCE = 20;
+    if (math.random(0,99) < CHANCE and killer:getMainJob() == JOB_BLU and killer:hasSpell(742) == false) then
+        killer:addSpell(SPELL_ID);
```
Comment (xdemolish) [inline scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua]: where is SPELL_ID defined?

File: scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua [hunk 5]
```diff
@@ -28,210 +39,144 @@ end;
...
+            end
+    elseif(mobHPP <= 98 and change == 0) then -- Chariots
+        mob:setModelId(1825);
+        mob:setLocalVar("change", 1);
+        for i = 1, 8 do
+            GetMobByID(petIDs[i]):setModelId(1820);
+            if petStatus[i] == 0 then
+               SpawnMob(petIDs[i]):updateEnmity(target);
+            end
+        end
```
Comment (xdemolish) [inline scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua]: strip out all the for loops inside each elseif, and set a var 'skin' to whatever you wanna change the pets' modelids to
then, outside the if/elseif block loop over the petids once and set the modelid tp 'skin' (whatever it was set in the if/elseif block)

**Comment by Loki79** [comment]
removed custom from PW

**Comment by ashdam** [comment]
Just curious. What's the meaning of PW prefix?

**Comment by Loki79** [comment]
Pandemonium Warden

**Comment by ashdam** [comment]
Thank you!

File: scripts/zones/AlTaieu/mobs/Jailer_of_Love.lua [hunk 6]
```diff
@@ -1,13 +1,137 @@
...
+        end
+    end
+end;
+
+-----------------------------------
+-- onMobDespawn
+-----------------------------------
+
+function onMobDespawn(mob)
+    SetServerVariable("JoL_Regen_Reduction", 0);
```
Comment (xdemolish) [inline scripts/zones/AlTaieu/mobs/Jailer_of_Love.lua]: shouldn't this be a localVar

**Comment by Loki79** [comment]
No it's a server var... and one i should commit with the rest of this i spode :P

**Comment by teschnei** [comment]
he's saying it SHOULD be a local var

server vars are only needed if it has to persist between spawn/despawn/server restart

**Comment by Loki79** [comment]
its tracked through JoL by pets killed and called in AV to set his regen amount.  Not sure i can do that in a local.

**Comment by xdemolish** [comment]
GetMobByID(JoL_ID):getLocalVar("regen_reduction")l

**Comment by TeoTwawki** [comment]
you can do getLocalVar() on a mob thats already dead? I take it those get cleared during the init or spawn for teh next pop rather than on death then.

**Comment by teschnei** [comment]
they're reset on spawn

**Comment by TeoTwawki** [comment]
Potential problem...Am I correct that localVar will reset on claim loss?

Was just looking over doing the convert for him (since is my fault they server var'd it told them just use that "for now" then forgot about it), and looking for any more non retail stuff was missed.

**Comment by teschnei** [comment]
no, it only resets on spawn

**Comment by TeoTwawki** [comment]
AV's regen reduction should remain even if claim resets. He has 2 reductions, 1 from JoL pets being slain and 1 from being hist by spells matching the day element. Just want to be sure I am not breaking that behavior.

**Comment by TeoTwawki** [comment]
...2hr stuff needs changed too. Legion has them random, but a few are missing and its set to go off per HP tier a limited number of times. Actually there's so much here that I think removing the AV script entirely is a better idea for now. Needs to have some spell list adjustments etc, gains access to meteor only below 80% OR during manafont. Sorry @Loki79 not blaming you Legion server just has to much stuff done to make it "go" without being retail accurate right now. 

I'll help get it cleaned up later and we can send AV in separately if dsp's ok with that?

**Comment by TeoTwawki** [comment]
Redid JoL's script to cal localvars tho

**Comment by TeoTwawki** [comment]
Did a bunch of cleanup, I think this is the best I can do with it on Legion's behalf.

**Comment by xdemolish** [comment]
still need that cleanup in JoL lua
kj can take a look at the rest :p

**Comment by TeoTwawki** [comment]
@xdemolish Like is aid on IRC, maybe its just because I am dead tired but I do not see how to keep it behaving retail with the suggested loop on JoJ. Hes only supposed to spawn 1 every 60 sec there, and can spawn up to 6 total. If I try to loop through it that way I am going to wind up doing extra crap anyway or else will wind up spawning all 6 at once, because I am no good with this. So somebody gonna have to explain how to keep it behaving as is, in detail, if you want that done. Everything else should be alright now and JoJ does work as written.

**Comment by xdemolish** [comment]
o ya
skim reading ftl

i'll have another look at it later today, think there were some other things which need changing

**Comment by xdemolish** [comment]
ya it can be loop'd loike
```
local spawn = false;
for mobid = mob:getID()+1, mob:getID()+6 do
    if spawn == true then
      return;
    else
       SpawnMob(mobid, 300):updateEnmity(target);
    end
end
```

**Comment by TeoTwawki** [comment]
@xdemolish You can't use a return there like that inside onMobFight

**Comment by xdemolish** [comment]
break w/e leave my lua alone

**Comment by TeoTwawki** [comment]
All commented on this have been handled now.

**Pull request merged**